portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,47 +36,71 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
 /** Servlet that returns comments and updates the datastore with comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
+    
+  private final Gson gson = new Gson();
   DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  UserService userService = UserServiceFactory.getUserService();
```

Comment:
Note that you should default fields in classes to `private final` unless you have specific reasons to do otherwise.